packages/api/src/external/carequality/process-subsequent-outbound-patient-discovery-resps.ts

```diff
@@ -0,0 +1,67 @@
+import { out } from "@metriport/core/util/log";
+import { Patient } from "@metriport/core/domain/patient";
+import { capture } from "@metriport/core/util/notifications";
+import { errorToString } from "@metriport/shared/common/error";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { makeOutboundResultPoller } from "../ihe-gateway/outbound-result-poller-factory";
+import { updatePatientDiscoveryStatus } from "./command/update-patient-discovery-status";
+import { getCQData } from "./patient";
+import { getPatientOrFail } from "../../command/medical/patient/get-patient";
+import { gatherXCPDGateways } from "./patient";
+import { getOutboundPatientDiscoveryResp } from "./command/outbound-resp/get-outbound-patient-discovery-resp";
+
+dayjs.extend(duration);
+
+const context = "cq.patient.subsequent.discover";
+const resultPoller = makeOutboundResultPoller();
+
+export async function processSubsequentOutboundPatientDiscoveryResps({
+  requestId,
+  patientId,
+  cxId,
+}: {
+  patientId: string;
+  requestId: string;
+  cxId: string;
+}): Promise<void> {
+  const baseLogMessage = `CQ PD Processing subsequent results - patientId ${patientId}`;
+  const { log } = out(`${baseLogMessage}, requestId: ${requestId}`);
+  const { log: outerLog } = out(baseLogMessage);
+
+  try {
+    const patient = await getPatientOrFail({ id: patientId, cxId });
+    const discoveryStatus = getCQData(patient.data.externalData)?.discoveryStatus;
+
+    if (discoveryStatus !== "processing") {
+      log(`Kicking off subsequent patient discovery`);
+      const leftoverGateways = await calculateLeftoverGateways(patient);
```

Comment:
This is quite expensive to be running on each response - although it would do so only until the status is updated to `processing` (could move the update PD status to be the first thing we do inside this `if`).

Wondering if we could make it so that if this executes, we trigger the lambda w/o `numOfGateways` and it would always finish upon timeout?

NABD, though

Reply:
Hmm i think it would probably overcomplicated the logic for race condition i think its probably fine to leave as is. Realistically this should only be ran once. Like when we update the discovery status shouldnt the patient then be locked? So the `getPatientOrFail` then needs to wait for the next request

Reply:
How is the patient locked? There's no DB tx around these commands.

Imagine like there's a sleep between each of those "await" commands and that sleep might mean something else gets executed for the same patient/requestId.

Instead of passing `undefined` for `numOfGateways` we could just pass `Number.MAX_SAFE_INTEGER` so it would always wait for the timeout (I think its fine on the second+ execution of the lambda to wait for the timetout, as there's no DQ scheduled to be executed). 

But we could move the call to `updatePatientDiscoveryStatus` to be the very first thing we do inside this `if`.

Reply:
I thought about doing this just thought it may have been a bit hacky but this makes sense